lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1723,12 +1729,180 @@ struct NodeBypass : public mlir::RewritePattern {
     return success();
   }
 };
+} // namespace
 
 void NodeOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                          MLIRContext *context) {
   results.insert<FoldNodeName>(context);
 }
 
+static SmallVector<Value> getCompleteWrite(Operation *lhs) {
+  auto lhsTy = lhs->getResult(0).getType();
+  if (!lhsTy.isa<BundleType>() && !lhsTy.isa<FVectorType>())
+    return {};
+
+  DenseMap<uint32_t, Value> fields;
+  for (Operation *user : lhs->getResult(0).getUsers()) {
+    if (user->getParentOp() != lhs->getParentOp())
+      return {};
+    if (auto aConnect = dyn_cast<StrictConnectOp>(user)) {
+      if (aConnect.getDest() == lhs->getResult(0))
+        return {};
+    } else if (auto subField = dyn_cast<SubfieldOp>(user)) {
+      for (Operation *subuser : subField.getResult().getUsers()) {
+        if (auto aConnect = dyn_cast<StrictConnectOp>(subuser)) {
+          if (aConnect.getDest() == subField) {
+            if (fields.count(subField.getFieldIndex())) // duplicate write
+              return {};
+            fields[subField.getFieldIndex()] = aConnect.getSrc();
+            continue;
+          }
+        }
```

Comment:
Rejecting all users except for connections might be too conservative. I believe we can allow most of operations which use the subfield as source. 
```
wire array: UInt<1>[2];
array[0] <= 0;
array[1] <= 0;
node a = array[0];
````
I agree that the current implementation is sufficient to clean up temporary wires for aggregate constants so
can we add TODO comment to improve that in the future?